modules/contact/src/actions/ContactAction.C

```diff
@@ -180,6 +187,9 @@ ContactAction::validParams()
       true,
       "Whether to generate the mortar mesh from the action. Typically this will be the case, but "
       "one may also want to reuse an existing lower-dimensional mesh prior to a restart.");
+  params.addParam<bool>("automatic_pairing_node_proximity",
```

Comment:
Per our discussion, I think it would be better to make this an enum, and have that be required if you're using the automatic pairing option. Maybe the parameter should be something like: `automatic_pairing_method = CENTROID | NODE`